python/whylogs/api/writer/whylabs.py

```diff
@@ -544,6 +537,36 @@ def _write_segmented_reference_result_set(self, file: SegmentedResultSet, **kwar
         else:
             return False, "Failed to upload all segments"
 
+    def _write_segmented_result_set(self, file: SegmentedResultSet, **kwargs: Any) -> Tuple[bool, str]:
+        """Put segmented result set for the specified dataset.
+
+        Parameters
+        ----------
+        file : SegmentedResultSet
+            SegmentedResultSet object representing the segmented result set for the specified dataset
+
+        Returns
+        -------
+        Tuple[bool, str]
+        """
+        # multi-profile writer
+        files = file.get_writables()
+        statuses: List[str] = list()
+        and_status: bool = True
+        if not files:
+            logger.warning("Attempt to write a result set with no writables, nothing written!")
+            return True, ""
+
+        logger.debug(f"About to write {len(files)} files:")
+        # TODO: special handling of large number of files, handle throttling
+        for view in files:
+            bool_status, status = self.write(file=view, **kwargs)
+            and_status = and_status and bool_status
+            statuses.append(status)
+        logger.debug(f"Completed writing {len(files)} files!")
+
+        return bool_status, ";".join(statuses)
```

Comment:
`bool_status` looks like it reflects only the status of the last write, this is surprising, I think you intended to return `and_status`?